presto-spi/src/main/java/io/prestosql/spi/predicate/Marker.java

```diff
@@ -182,6 +182,14 @@ public boolean isAdjacent(Marker other)
         if (isUpperUnbounded() || isLowerUnbounded() || other.isUpperUnbounded() || other.isLowerUnbounded()) {
             return false;
         }
+        // Allow merging two consecutive long values into a range:
+        if (bound == Bound.EXACTLY && other.bound == Bound.EXACTLY && type.getJavaType() == long.class) {
```

Comment:
same comment on `javaType == long.class` condition as in https://github.com/trinodb/trino/pull/3316#discussion_r681715134